src/unsafe-rust/unsafe-traits.md

```diff
@@ -8,36 +8,35 @@ Like with functions, you can mark a trait as `unsafe` if the implementation must
 guarantee particular conditions to avoid undefined behaviour.
 
 For example, the `zerocopy` crate has an unsafe trait that looks
-[something like this](https://docs.rs/zerocopy/latest/zerocopy/trait.AsBytes.html):
+[something like this](https://docs.rs/zerocopy/latest/zerocopy/trait.IntoBytes.html):
 
 ```rust,editable
-use std::mem::size_of_val;
-use std::slice;
+use std::{mem, slice};
 
 /// ...
 /// # Safety
 /// The type must have a defined representation and no padding.
-pub unsafe trait AsBytes {
-    fn as_bytes(&self) -> &[u8] {
-        unsafe {
-            slice::from_raw_parts(
-                self as *const Self as *const u8,
-                size_of_val(self),
-            )
-        }
+pub unsafe trait IntoBytes {
+    fn as_bytes(&self) -> &[u8]
+    where
+        Self: Immutable,
+    {
+        let len = mem::size_of_val(self);
+        let slf: *const Self = self;
+        unsafe { slice::from_raw_parts(slf.cast::<u8>(), len) }
     }
 }
 
-// SAFETY: `u32` has a defined representation and no padding.
-unsafe impl AsBytes for u32 {}
+/// Types which are free from interior mutability.
+pub unsafe trait Immutable {}
```

Comment:
@mgeisler, I think it had to be added to make the tests pass.

Reply:
Shouldn't we still have a `SAFETY` comment here, given the `unsafe impl`?

Reply:
> EDIT: Oh, I misread this change. It would be ideal for this slide to still actually demonstrate an `unsafe impl` as that is the unsafe capability being discussed here.

Yeah, I agree we should keep the `unsafe impl`.

@semihbkgr, could you simplify the `IntoBytes` trait and remove the `where Self: Immutable` bound? The code has to be super simple and just demonstrate the concept of an unsafe trait (it should not be 100% like what zerocopy has).